idaes/ui/fsvis/static/js/paper.js

```diff
@@ -63,6 +63,28 @@ export class Paper {
         // context menu
         self._paper.on("element:contextmenu", function(cellView, evt) {
             cellView.model.rotate(90)
+            // This is needed to keep the text labels for the unit models in the correct orientation
+            // x and y were specifically picked to keep the label in the same place 
+            // in relation to the unit model (bottom middle)
+            let x = 0
+            let y = 0
+            if (cellView.model.angle() == 0) {
+                x = 0
+                y = 5
+            }
+            else if (cellView.model.angle() == 90) {
```

Comment:
Is the angle always guaranteed to be positive and "normalized", i.e. in the interval `[0, 360)`? Otherwise, it would require some pre-processing to make sure that e.g. 270 is not missed because it is returned as -90.

Reply:
Yes the angle is always positive and normalized.